test/e2e/page-objects/pages/confirm-tx-page.ts

```diff
@@ -0,0 +1,40 @@
+import { Driver } from '../../webdriver/driver';
+
+class ConfirmTxPage {
+  private driver: Driver;
+
+  private confirmButton: string;
+
+  private editButton: string;
+
+  private transactionFee: string;
+
+  constructor(driver: Driver) {
+    this.driver = driver;
+    this.confirmButton = '[data-testid="page-container-footer-next"]';
+    this.editButton = '[data-testid="confirm-page-back-edit-button"]';
+    this.transactionFee = '[data-testid="confirm-gas-display"]';
+  }
+
+  async check_pageIsLoaded(): Promise<void> {
+    try {
+      await this.driver.waitForSelector(this.editButton);
+      await this.driver.waitForSelector(this.confirmButton);
+      await this.driver.waitForSelector(this.transactionFee);
```

Comment:
I am thinking if this could introduce a race condition, since having the element present in the DOM doesn't mean that ie its value is currently loaded, or the button is enabled etc.


I think for the buttons is fine, since we already have the guards for being buttons enabled in the click methods, however it could bring confusion with the transactionFee value or the total amount value if we add it.

Ie, in occasions we display a tx fee value, which is then updated after few seconds. So I wonder what makes a page 'loaded' :thinking: 
I think this depends on each screen, so we might need to make it slightly different case by case: in confirmation's page, we would consider it fully "loaded" when:
- we have the correct total amount 
- the correct fees expected
- the confirm button enabled (redundant with the click method, but ok to have)
So in this case, we could pass both values to this function from each test

`async check_pageIsLoaded(totalExpectedAmount, totalExpectedFees)`
and then look for the selector with the value passed.

This will be really good, since will mitigate all the race conditions related to this issue in the page level and ensure no more occurrences are introduced in the tests using this page

Reply:
Yes, i agree confirmation page is a special case that need more check to mitigate all race condition issues. I adjusted as you suggested ! Thanks !